apps/chat/src/components/Marketplace/ApplicationDetails/ApplicationDetails.tsx

```diff
@@ -49,20 +57,26 @@ const ApplicationDetails = ({ onClose, entity }: Props) => {
   }, [entities, entity.name]);
 
   const handleUseEntity = useCallback(() => {
-    selectedConversations.forEach((conv) =>
-      dispatch(
-        ConversationsActions.updateConversation({
-          id: conv.id,
-          values: {
-            ...getConversationModelParams(
-              conv,
-              entity.id,
-              modelsMap,
-              addonsMap,
-            ),
-          },
-        }),
-      ),
+    const conversationToApplyModel =
+      selectedConversations.find((conv) =>
+        compareIdWithQueryParamId(
+          conv.id,
+          searchParams.get(MarketplaceQueryParams.fromConversation),
```

Comment:
read it once and save in constant

Reply:
this is calculated and applied once anyway, when the use model is clicked.